pennylane/ops/qubit/observables.py

```diff
@@ -435,13 +435,20 @@ class BasisStateProjector(Projector, Operation):
     # The call signature should be the same as Projector.__new__ for the positional
     # arguments, but with free key word arguments.
     def __init__(self, state, wires, id=None):
-        wires = Wires(wires)
-        state = list(qml.math.toarray(state).astype(int))
+        wires = qml.wires.Wires(wires)
 
-        if not set(state).issubset({0, 1}):
-            raise ValueError(f"Basis state must only consist of 0s and 1s; got {state}")
+        if state is None:
+            raise ValueError("Basis state cannot be None")
 
-        super().__init__(state, wires=wires, id=id)
+        if not qml.math.is_abstract(state):
+            state_array = qml.math.toarray(state)
+            state_array = state_array.astype(int)
+            if not qml.math.isininterval(state_array, 0, 1).all():
```

Comment:
I can't seem to find documentation for `numpy.isininterval` or an `isininterval` in any of the ML frameworks. Mind linking some documentation to it? Would this be a custom thing we would need to add to `pennylane/math`?

Reply:
Yeah, i understand your confusion. I was working with some custom methods that i created in `pennylane.math`, then i forgot to undo the changes and rollback completely. Like deleted those custom methods in  `pennylane.math` but forgot to change it in `observables.py`, mostly because the IDE that i was using didnt mark anything. Rectified those silly typos in the recent commits. Thanks for pointing out.